curriculum/challenges/english/25-front-end-development/lecture-understanding-core-javascript-fundamentals/6732c6d4dec34c61850a1276.md

```diff
@@ -8,7 +8,41 @@ dashedName: what-is-memory-management-and-how-does-it-work-in-javascript
 
 # --description--
 
-Watch the lecture video and answer the questions below.
+Watch the video or read the transcript and answer the questions below.
+
+# --transcript--
+
+What is memory management, and how does it work in JavaScript?
+
+Memory management is an essential concept in programming, but it can be a bit confusing for beginners, let's break it down in simple terms.
+
+When you run a program including JavaScript code in a web browser it needs memory to store all the information its working with, this includes varibles, functions, objects, basically everything your code creates and uses. Memory management is the process of controlling this memory, allocating it when needed, and freeing it up when it's no longer needed. In some programming languages, developers have to manually manage memory. They need to explicitly tell the computer when to allocate memory for new things and when to free up memory that's no longer needed. This can be powerful but also tricky as for getting to free memory can lead to memory leaks.
```

Comment:
```suggestion
When you run a program, including JavaScript code in a web browser, it needs memory to store all the information its working with. This includes varibles, functions, objects, basically everything your code creates and uses. Memory management is the process of controlling this memory, allocating it when needed, and freeing it up when it's no longer needed. In some programming languages, developers have to manually manage memory. They need to explicitly tell the computer when to allocate memory for new things and when to free up memory that's no longer needed. This can be powerful but also tricky as for getting to free memory can lead to memory leaks.
```